llvm/test/CodeGen/AMDGPU/sdwa-peephole-cndmask-fail.ll

```diff
@@ -0,0 +1,40 @@
+; RUN: llc %s -march=amdgcn -mcpu=gfx1030 -o - 2>&1 | FileCheck %s
+
+; In this test, V_CNDMASK_B32_e64 gets converted to V_CNDMASK_B32_e32,
+; but the expected conversion to SDWA does not occur.  This led to a
+; compilation error, because the use of $vcc in the resulting
+; instruction must be fixed to $vcc_lo for wave32 which only happened
+; after the full conversion to SDWA.
+
+
+; CHECK-NOT: {{.*}}V_CNDMASK_B32_e32{{.*}}$vcc
+; CHECK-NOT: {{.*}}Bad machine code: Virtual register defs don't dominate all uses
+; CHECK: {{.*}}v_cndmask_b32_e32{{.*}}vcc_lo
+
+define amdgpu_kernel void @quux(i32 %arg, i1 %arg1, i1 %arg2) {
```

Comment:
This can probably be reduced a lot, and can probably stop being a kernel

Reply:
This went trough llvm-reduce. I cannot reduce it signficantly by hand either. In particular, the control flow seems to be necessary. It doesn't need to be a kernel.